recipes/mingw-w64/all/conanfile.py

```diff
@@ -13,29 +13,193 @@ class MingwConan(ConanFile):
     settings = "os", "arch", "compiler"
     options = {"threads": ["posix", "win32"], "exception": ["seh", "sjlj"]}
     default_options = {"threads": "posix", "exception": "seh"}
-    build_requires = "7zip/19.00"
     no_copy_source = True
 
     def validate(self):
-        if self.settings.os != "Windows":
-            raise ConanInvalidConfiguration("MinGW is only supported by Windows.")
+        valid_os = self.conan_data["sources"][self.version]["url"].keys()
+        if str(self.settings.os) not in valid_os:
+            raise ConanInvalidConfiguration(f"MinGW {self.version} is only supported for the following operating systems: {valid_os}")
```

Comment:
Don't use f-strings (yet).
We're trying to not use "modern" python features.

Reply:
Will be fixed in upcoming commit.